iped-carvers/iped-carvers-impl/src/main/java/iped/carvers/standard/AbstractCarver.java

```diff
@@ -212,10 +235,40 @@ public void notifyHit(IItem parentEvidence, Hit hit) throws IOException {
         }
 
         clearOldHeaders(parentEvidence);
+        processHeadersWithStopOnNext(parentEvidence, hit.getOffset());
+    }
+
+    protected void processHeadersWithStopOnNext(IItem parentEvidence, long currOffset) throws IOException {
+        if (headersWithStopOnNext.isEmpty()) {
+            return;
+        }
+        Hit last = headersWithStopOnNext.getLast();
+        Iterator<Hit> it = headersWithStopOnNext.iterator();
+        while (it.hasNext()) {
+            Hit hit = it.next();
+            CarverType type = hit.getSignature().getCarverType();
+            long len = Math.min(parentEvidence.getLength() - hit.getOffset(), type.getMaxLength());
+            boolean ready = false;
+            if (hit.getOffset() + len <= currOffset) {
+                ready = true;
+            } else if (!hit.equals(last) && type.equals(last.getSignature().getCarverType())) {
+                ready = true;
+                len = Math.min(len, last.getOffset());
```

Comment:
Hi @tc-wleite. I processed a 2.2M items case in the afternoon and carving results seem fine, no regression found. Taking a look at the code, not sure, but shouldn't this line be `len = Math.min(len, last.getOffset() - hit.getOffset());`? I didn't test with Shareaza SD files to test this new logic.

Reply:
Yes! Sorry about that!
Most of the hits here were in offset zero, but I will repeat my tests fixing this line.

Reply:
Pushed commit [21418f4](https://github.com/sepinf-inc/IPED/pull/1826/commits/21418f4bd8c41287da870b5bbf779f388cf99d8f) fixing it.
Repeated the tests with the SDs I have here and it seems fine now.
In the case where 3 SDs were recovered from the same file, the first was in offset 0, so the result was the same. The third (last) is not affected. But the second item was wrong before (didn't stop where the 3rd started), and now it is fine.

Thanks for catching this!!

Reply:
Great! Thank you for this PR!